packages/number-field/test/value-commit.test.js

```diff
@@ -0,0 +1,251 @@
+import { expect } from '@esm-bundle/chai';
+import { fixtureSync, nextRender, outsideClick } from '@vaadin/testing-helpers';
+import { sendKeys } from '@web/test-runner-commands';
+import sinon from 'sinon';
+import '../src/vaadin-number-field.js';
+
+describe('value commit', () => {
+  let numberField, valueChangedSpy, validateSpy, changeSpy;
+
+  function expectNoValueCommit() {
+    expect(valueChangedSpy).to.be.not.called;
+    expect(validateSpy).to.be.not.called;
+    expect(changeSpy).to.be.not.called;
+  }
+
+  function expectValueCommit(value) {
+    expect(valueChangedSpy).to.be.calledOnce;
+    // TODO: Optimize the number of validation runs.
+    expect(validateSpy).to.be.called;
+    expect(validateSpy.firstCall).to.be.calledAfter(valueChangedSpy.firstCall);
+    expect(changeSpy).to.be.calledOnce;
+    expect(changeSpy.firstCall).to.be.calledAfter(validateSpy.firstCall);
+    expect(numberField.value).to.equal(value);
+  }
+
+  function expectValidationOnly() {
+    expect(valueChangedSpy).to.be.not.called;
+    expect(validateSpy).to.be.calledOnce;
+    expect(changeSpy).to.be.not.called;
+  }
+
+  beforeEach(async () => {
+    numberField = fixtureSync(`<vaadin-number-field></vaadin-number-field>`);
+    await nextRender();
+    validateSpy = sinon.spy(numberField, 'validate').named('validateSpy');
+
+    valueChangedSpy = sinon.spy().named('valueChangedSpy');
+    numberField.addEventListener('value-changed', valueChangedSpy);
+
+    changeSpy = sinon.spy().named('changeSpy');
+    numberField.addEventListener('change', changeSpy);
+
+    numberField.focus();
+  });
+
+  describe('default', () => {
+    it('should not commit but validate on blur', () => {
+      numberField.blur();
+      expectValidationOnly();
+    });
+
+    it('should not commit on Enter', async () => {
+      await sendKeys({ press: 'Enter' });
+      expectNoValueCommit();
+    });
+
+    it('should not commit on Escape', async () => {
+      await sendKeys({ press: 'Escape' });
+      expectNoValueCommit();
+    });
+  });
+
+  describe('entering parsable input', () => {
+    beforeEach(async () => {
+      await sendKeys({ type: '1' });
+    });
+
+    it('should commit on blur', () => {
+      numberField.blur();
+      expectValueCommit('1');
+    });
+
+    it('should commit on Enter', async () => {
+      await sendKeys({ press: 'Enter' });
+      expectValueCommit('1');
+    });
+  });
+
+  describe('entering unparsable input', () => {
+    beforeEach(async () => {
+      await sendKeys({ type: '-' });
+    });
+
+    it('should not commit by default', () => {
+      expectNoValueCommit();
+    });
+
+    it('should not commit but validate on blur', () => {
+      numberField.blur();
+      expectValidationOnly();
+      expect(numberField.inputElement.validity.badInput).to.be.true;
+    });
+
+    it('should not commit but validate on Enter', async () => {
```

Comment:
This test currently fails due to https://github.com/vaadin/web-components/issues/5113

Reply:
After an internal discussion, we've decided to skip these tests for now. The issue will be fixed later.